petab/core.py

```diff
@@ -38,11 +38,15 @@ class Problem:
     """
 
     def __init__(self,
-                 sbml_file,
-                 condition_file,
-                 measurement_file,
+                 sbml_file=None,
```

Comment:
I see here two possible issues (correct me if wrong):
* The ``__getstate__`` and ``__setstate__`` methods only serialize file names, not the (possibly expensive) datasets. These methods are requried for parallelization. But I see the point of not requiring underlying files, but only dataframes.
* The double passing of _files and _dfs makes it unintuitive whether file names of dfs are used.

What I would suggest: In ``__init__``, only pass the dfs (these need to be kept then in ``__setstate__``, but that might be unavoidable). Second, introduce a static method ``from_files``, where the files are taken and dfs created. This could be similar to ``from_folder``, and the latter could also call the former directly.

Reply:
Sounds reasonable, generally aggreeing. I see that pickling dfs is expensive, on the other hand it is a priori not clear, that the files will exist on the receiver side in the same location. Furthermore the files might have changed in between. Same goes for the dfs which could have been changed since loading. Maybe the failsafe default should be pickling dataframes which is the real state and make it optional (and the users responsibility ensure same state) to pickle only file names.

Reply:
Yes, totally agree, in particular the files could not exist on target system thing. So let's maybe just have the dfs in the PetabProblem class, and only use file names to construct them in some static method. For our problems, even large scale problems seem to have less than 100MB files, so that should not be a problem.
Or is there a reason why the file names should be remembered?

Reply:
In the longer run I thought about being able to load them, modify them and save them back all at once from within that class. However, I am not sure if that would be a common use case. Passing file names in that case wouldn't be too hard...
So, agreeing on taking them out.

Reply:
Ah ok. Also sounds good of course, e.g. when the sbml model needs to be changed, or some missing columns values attachted to the tables. In these cases a passed file name would be required though.

Another point: The sbml model cannot be serialized, so this is only possible via the file name :( .

Reply:
Well, it can... as string... Not that big a deal...

Reply:
Done.